accepted/single-file/design.md

```diff
@@ -66,225 +32,254 @@ Publishing to a single file can be triggered by adding the following property to
 
-* The MSBuild logic in `dotnet SDK` should be crafted such that [IlLinker](https://github.com/dotnet/core/blob/master/samples/linker-instructions.md), [crossgen](https://github.com/dotnet/coreclr/blob/master/Documentation/building/crossgen.md), and the single-file bundler run in that order in the build/publish sequence. 
-* External tools like [Fody](https://github.com/Fody/Fody) that use  `AfterBuild`/`AfterPublish` targets may need to adapt to expect the significantly different output generated by publishing to a single file. The goal in this case is to provide sufficient documentation and guidance.
+### Interaction with External Tools
 
-### The Host
+Once the single-file-publish tooling is added to the publish pipeline, other static binary transformation tools may need to adapt its presence. For example, tools like [Fody](https://github.com/Fody/Fody) that use  `AfterBuild`/`AfterPublish` targets may need to adapt to expect the significantly different output generated by publishing to a single file. The goal in this case is to provide sufficient documentation and guidance.
 
-On Startup, the AppHost checks if it has embedded files. If so, it 
+## The Bundler
 
-* Memory maps the entire bundle file.
-* Extracts the necessary files (ex: native binaries) to disk (if any) as explained in this [document](extract.md). 
-* Sets up data-structures so other components can access files embedded directly, as explained in this [document](direct.md)
+The bundler is a tool that embeds the managed app and its dependencies into the native `AppHost` executable. The functional details of the bundler are explained in [this document](bundler.md). 
 
-#### Dependency Resolution
+## The Host
+
+### Startup
+
+On Startup, the [host components](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-components.md) perform the following functions:
+
+* **AppHost**: The AppHost identifies itself as a single-file bundle (by checking the [bundle marker](bundler.md#bundle-marker)) before invoking [HostFxr](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-components.md#host-fxr).
+
+* **HostFxr**: If invoked from a single-file app, HostFxr process the `runtimeconfig.json` and `deps.json` files directly from the bundle. The location of these `json` files  are identified directly from the bundle-header for simple access.
+
+* **HostPolicy**: Much of the bundle-processing is performed within HostPolicy.
+  
+  * If the app needs extraction, extracts out the appropriate files as explained in [this document](extractor.md).
+  
+  * Reads the `deps.json` file directly from the bundle and resolves dependencies.
```

Comment:
The HostPolicy probing can continue as if the contents of the bundle were available alongside the app (using `bundle_probe` instead of file probing. I did not elaborate on this yet, since `deps.json` probing itself is still under discussion. But I agree that this requires more explanation.